server/text_generation_server/utils/layers.py

```diff
@@ -687,7 +687,7 @@ def load(cls, prefix, weights, eps=1e-6):
             weight = weights.get_tensor(f"{prefix}.weight")
             return cls(weight, eps)
 
-        def forward(self, hidden_states, residual=None):
+        def forward(self, hidden_states, residual=None, force_downcast_after=False):
```

Comment:
I'd personally use a different method `forward_downcast_after`.

Also this only triggers when hidden_size > 8192 meaning it won't trigger for gemma.
Having a dedicated method seems simpler.

Reply:
that makes alot of sense thanks for the comments. I've removed the branching logic from `FastRMSNorm` and added `forward_downcast_after` into the gemma modeling code